File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    def sol(n):
        count = 0
        for i in range(1,n):
            if n % i == 0:
                count += 1
        if count % 2:
            return True
        else:
            return False

    for i in range(left, right+1):
        if sol(i):
            answer += i
        else:
            answer -= i

    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int measure(int num){
        int count = 0;
        for (int i = 1; i<=num; i++){
            if(num % i == 0){
                count++;
            }
        }
        
        if(count % 2 == 0){
            return num;
        }else{
            return -num;
        }
    }
    
    public int solution(int left, int right) {
        int answer = 0;
        for (int i = left; i<= right; i++){
            answer += measure(i);
        }
        return answer;
    }
}